### PR TITLE
catching all exception from the API (circles+shares)

### PR DESCRIPTION
**Pull request by ArtificialOwl**

### CODE REVIEW

**Review by ChristophWurst — APPROVED** [review]

**Review by kesselb — APPROVED** [review]

**Comment by kesselb** [comment]
I would suggest to also change it here: https://github.com/nextcloud/server/blob/ac8939a1a81c9d427ee5fe2f50517324e76a726f/apps/dav/lib/Connector/Sabre/Principal.php#L478-L484

Found it a few days ago while looking into https://github.com/nextcloud/server/pull/23652.

**Comment by MorrisJobke** [comment]
@daita Keep in mind to do PRs in the future against master and then we use the backport-bot (via `/backport to stable20`) to backport to other branches.

File: apps/files_sharing/lib/Controller/ShareAPIController.php [hunk 1]
```diff
@@ -1619,7 +1620,8 @@ private function shareProviderResharingRights(string $userId, IShare $share, $no
 					return true;
 				}
 				return false;
-			} catch (QueryException $e) {
+			} catch (Exception $e) {
+				\OC::$server->getLogger()->logException($e, ['app' => 'circles']);
```
Comment (MorrisJobke) [inline apps/files_sharing/lib/Controller/ShareAPIController.php:1624]: And also this should be injected in the constructor.

**Comment by daita** [comment]
closing this one, as the maintenance of the shares entries is now managed by the app itself.

**Pull request closed without merging**